### PR TITLE
Update architect orb and fix deployment

### PR DESCRIPTION
**Pull request by tfussell**
Towards https://github.com/giantswarm/giantswarm/issues/8436

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -2,7 +2,7 @@
 version: 2.1
 
 orbs:
-  architect: giantswarm/architect@0.8.1
+  architect: giantswarm/architect@dev:deploy-without-tag
```
Comment (tfussell) [inline .circleci/config.yml]: Will update to `0.8.3` once https://github.com/giantswarm/architect-orb/pull/103 is merged.

File: .circleci/config.yml [hunk 2]
```diff
@@ -36,21 +36,25 @@ jobs:
       - checkout
       - run:
           name: Create chart directory
-          command: mkdir -p helm/<<parameters.app_name>>/templates
+          command: mkdir -p helm/<< parameters.app_name >>/templates
```
Comment (tfussell) [inline .circleci/config.yml]: Trying to be consistent with spacing around parameters.

File: .circleci/config.yml [hunk 3]
```diff
@@ -36,21 +36,25 @@ jobs:
...
       - run:
           name: Copy template release into chart templates
-          command: mv .circleci/release.yaml helm/<<parameters.app_name>>/templates
+          command: mv .circleci/release.yaml helm/<< parameters.app_name >>/templates
       - run:
           name: Copy releases to values.yaml
           command: |
-            echo "versions:" > helm/<<parameters.app_name>>/values.yaml
-            sed 's/^/  /' aws.yaml >> helm/<<parameters.app_name>>/values.yaml
```
Comment (tfussell) [inline .circleci/config.yml]: I overlooked this in the last PR. This should have been a parameter.

File: .circleci/config.yml [hunk 4]
```diff
@@ -36,21 +36,25 @@ jobs:
...
       - run: 
           name: Create Chart.yaml
           command: |
-            cat \<<EOF > helm/<<parameters.app_name>>/Chart.yaml
-            name: <<parameters.app_name>>
+            cat \<<EOF > helm/<< parameters.app_name >>/Chart.yaml
+            apiVersion: v1
+            name: << parameters.app_name >>
             version: [[ .Version ]]
+            appVersion: master
```
Comment (tfussell) [inline .circleci/config.yml]: Required by helm linter along with `description` and `home`. Not sure what this one should be.

File: .circleci/config.yml [hunk 5]
```diff
@@ -74,6 +78,7 @@ workflows:
           app_catalog_test: control-plane-test-catalog
           attach_workspace: true
           chart: releases-kvm
+          production_branch: master
```
Comment (tfussell) [inline .circleci/config.yml]: This means that we should push to `kvm-app-collection` on merge to `master`.

File: .circleci/release.yaml [hunk 6]
```diff
@@ -3,7 +3,7 @@ apiVersion: release.giantswarm.io/v1alpha1
 kind: Release
 metadata:
   annotations:
-    giantswarm.io/docs: https://pkg.go.dev/github.com/giantswarm/apiextensions/pkg/apis/release/v1alpha1?tab=doc#Release
+    giantswarm.io/docs: https://docs.giantswarm.io/reference/cp-k8s-api/releases.release.giantswarm.io/
```
Comment (tfussell) [inline .circleci/release.yaml]: This was made public today so I just wanted to update to the correct URL.

File: .circleci/config.yml (hunk 4 above)
Comment (MarcelMue) [reply .circleci/config.yml]: I think this is meant just for apps but master is accurate as it's an unique set of CRs coming from master.

File: .circleci/release.yaml (hunk 6 above)
Comment (MarcelMue) [reply .circleci/release.yaml]: Nice :heart:

File: .circleci/config.yml [hunk 7]
```diff
@@ -2,7 +2,7 @@
 version: 2.1
 
 orbs:
-  architect: giantswarm/architect@0.8.1
+  architect: giantswarm/architect@0.8.3
```
Comment (tfussell) [inline .circleci/config.yml]: Adds the `on_tag` parameter.

File: .circleci/release.yaml (hunk 6 above)
Comment (rossf7) [reply .circleci/release.yaml]: +1 ❤️

**Pull request merged**